### ISSUE TITLE
Cannot connect to mixer

### ISSUE THREAD

**Issue by NaturalDevCR** [fmalcher/soundcraft-ui#153]
Hi, I'm just trying to work again in my project, which uses your library, but the connection is failing.

- I'm sure the IP is correct
- The mixer is on, and I'm connected to the mixer's wifi
- I can load the mixer through its ip on the browser with no problems.

The connection subscriber shows the following error:

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/9f53aaac-e307-4751-b822-876836443c5d">

This is my code:

Dependencies:

```
"dependencies": {
    "@quasar/extras": "^1.16.12",
    "@vueuse/core": "^11.1.0",
    "is-ip": "^5.0.1",
    "pinia": "^2.2.4",
    "pinia-plugin-persistedstate": "^3.2.3",
    "quasar": "^2.17.0",
    "soundcraft-ui-connection": "^2.7.0",
    "vue": "^3.5.12",
    "vue-country-flag-next": "^2.3.2",
    "vue-i18n": "^10.0.4",
    "vue-router": "^4.4.5",
    "vue3-slider": "^1.10.1",
    "webmidi": "^3.1.11"
  },
```

mixer-store.ts:

```
async uiConnect() {
      if (isValidMixerIp(this.ip)){
        if (isAllowedURL(this.ip)){
          this.isDemoMode = true
        }else {
          this.isDemoMode = false
          console.log(this.ip)
          this.conn = new SoundcraftUI(this.ip)
          this.conn.status$.subscribe((status:any) => {
            console.log(status)
            this.connStatus = status.type
          });
          console.log(this.conn)
          try {
            await this.conn.connect()
          } catch (e) {
            console.log(JSON.stringify(e))
          }
          this.getUiModel()
          this.getFirmwareVersion()
        }
      }else {
        this.setupModal = true
      }

    },
```

I'm I doing something wrong?, I just checked the documentation and I don't see anything wrong in my code.

**Comment by fmalcher** [comment]
Before we think about the inner details, have you tried deleting `node_modules` and `package-lock.json` and do a clean install?

**Comment by fmalcher** [comment]
I cloned your project and got the same error. I never saw this problem before and it seems like nobody else is having it. I suspect a weird combination of environments, maybe quasar who brings an own abstraction around WebSocket – who knows.

I tried to get closer to the root of the problem. The object that is used as `WebSocketCtor` can indeed not be called as a constructor. This comes from the `isomorphic-ws` library which cares about platform independence for WebSocket. If I just put `WebSocket` in there without the lib, it works in your project. This makes me guess that quasar is the problem here.

Can you try to reproduce this in a minimal project? Just a new quasar project, latest version of soundcraft-ui-connection and then try to establish a connection in the main file. If you can provide this, this would help me to get closer to what causes the confusion.

I also think about not providing a WebSocket abstraction within the library but instead, let the users pass in their desired WS object. I have to think about this!

**Comment by NaturalDevCR** [comment]
It's weird because the version published in github hasn't updated in a while, and it was working with the versions that were available at that moment, unless there's something new with the browser or some deprecated thing...

But I'll try with a minimal installation, and see if Quasar is the problem

Thanks a lot for digging in.

**Comment by NaturalDevCR** [comment]
So after using a minimal quasar installation, same error happened:

<img width="1102" alt="image" src="https://github.com/user-attachments/assets/54310651-e0ce-4aa3-97eb-f7901a6b70a5">


I Guess I'll move to a different framework, I love Quasar because integrates very nicely multiplatform, before I close this issue I'd like to try with vuejs itself and see if everything goes well just to discard is a vuejs imcompatibility.

**Comment by fmalcher** [comment]
I released version 3.0.0-beta.1 with the new WebSocket abstraction.
What I said in the other issue is wrong: `modern-isomorphic-ws` does not solve the problem. I just forgot to NOT bundle the package with the library, so it was downleveled to the browser and used the `WebSocket` object.

I just tried this with your project and it works:

```ts
// mixer-store.ts
this.conn = new SoundcraftUI({
  targetIP: this.ip,
  webSocketCtor: WebSocket,
});
```

So in the end, letting people pass in their own WebSocket constructor seems to have solved the problem 😄 
Can you please try this out, too?

**Comment by NaturalDevCR** [comment]
> I released version 3.0.0-beta.1 with the new WebSocket abstraction. What I said in the other issue is wrong: `modern-isomorphic-ws` does not solve the problem. I just forgot to NOT bundle the package with the library, so it was downleveled to the browser and used the `WebSocket` object.
> 
> I just tried this with your project and it works:
> 
> ```ts
> // mixer-store.ts
> this.conn = new SoundcraftUI({
>   targetIP: this.ip,
>   webSocketCtor: WebSocket,
> });
> ```
> 
> So in the end, letting people pass in their own WebSocket constructor seems to have solved the problem 😄 Can you please try this out, too?

It works perfectly, thanks a lot!

**Comment by fmalcher** [comment]
v3.0.0 is now released
